commodore/dependency_mgmt.py

```diff
@@ -84,27 +84,43 @@ def _discover_components(cfg, inventory_path):
     return sorted(components), component_aliases
 
 
-def _read_component_urls(cfg: Config, component_names) -> Dict[str, str]:
+def _read_components(
+    cfg: Config, component_names
+) -> Tuple[Dict[str, str], Dict[str, Optional[str]]]:
     component_urls = {}
+    component_versions = {}
 
-    if cfg.debug:
-        click.echo(f"   > Read commodore config file {cfg.config_file}")
-    try:
-        commodore_config = yaml_load(cfg.config_file)
-    except Exception as e:
-        raise click.ClickException(
-            f"Could not read Commodore configuration: {e}"
-        ) from e
+    reset_reclass_cache()
+    inv = inventory_reclass(cfg.inventory.inventory_dir)
+    cluster_inventory = inv["nodes"][cfg.inventory.bootstrap_target]
+    components = cluster_inventory["parameters"].get("components", None)
+    if not components:
+        raise click.ClickException("Component list (parameters.components) missing")
 
-    for component in commodore_config.get("components", []):
-        if component["name"] in component_names:
-            component_urls[component["name"]] = component["url"]
+    for component_name in component_names:
+        if component_name not in components:
+            raise click.ClickException(f"Unknown component '{component_name}'")
 
-    for name in component_names:
-        if name not in component_urls:
-            raise click.ClickException(f"No url for component '{name}' configured")
+        info = components[component_name]
 
-    return component_urls
+        if "url" not in info:
+            raise click.ClickException(
+                f"No url for component '{component_name}' configured"
+            )
+
+        component_urls[component_name] = info["url"]
+        if cfg.debug:
+            click.echo(f" > URL for {component_name}: {component_urls[component_name]}")
+        if "version" in info:
+            component_versions[component_name] = info["version"]
+        else:
+            component_versions[component_name] = "master"
```

Comment:
Would it be possible to use `origin/HEAD` instead of hard coding `master` as the branch name?

Reply:
We might even think about making the version required instead of defaulting to `master`. This would require people to think about the version they want to install? Or maybe log a warning and at some point make it required?

Reply:
> Would it be possible to use `origin/HEAD` instead of hard coding `master` as the branch name?

I've already got a follow-up PR for using `origin/HEAD` by default instead of master. I decided to split the two to keep the review simpler.

> We might even think about making the version required instead of defaulting to `master`. This would require people to think about the version they want to install? Or maybe log a warning and at some point make it required?

That might be a good idea, especially since we're considering moving to versioned components.

Implementation-wise, I was thinking about building some "deprecation notice" framework yesterday. I was thinking about warning about uses of `parameters.component_versions`, but deprecating components without versions might be a good use case too.